src/processor/tree/tree_wrapper.rs

```diff
@@ -65,19 +67,19 @@ impl TreeWrapper {
                 .expect("invalid key index");
             let value = self.process_value(key, *value);
 
-            key_value_pairs.push((key, value));
+            key_values.push(TreeEntry::new(key, index, value));
         }
 
-        let output = self.tree.extend(key_value_pairs);
+        let output = self.tree.extend(key_values);
         let root_hash = output.root_hash;
 
-        assert_eq!(root_hash.as_bytes(), block.new_state_root);
         tracing::debug!(
             "Root hash of block {} = {}",
             block.l2_block_number,
             hex::encode(root_hash)
         );
 
+        assert_eq!(root_hash.as_bytes(), block.new_state_root);
```

Comment:
Just as a note for the future: we should catch the error here and reset the tree to the latest version instead